app/next-client-app/components/scanreports/DeleteDialog.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "../ui/dialog";
+import { toast } from "sonner";
+import { Button } from "../ui/button";
+import { deleteScanReport } from "@/api/scanreports";
+import { DialogTrigger } from "@radix-ui/react-dialog";
+import { ReactNode } from "react";
+import { useRouter } from "next/navigation";
+
+interface DeleteDialogProps {
+  id: number;
+  redirect?: boolean;
+  isOpen?: boolean;
+  setOpen?: (isOpen: boolean) => void;
+  children?: ReactNode | ReactNode[];
+}
+
+const DeleteDialog = ({
+  id,
+  redirect = false,
+  isOpen,
+  setOpen = () => {},
+  children,
+}: DeleteDialogProps) => {
+  const router = useRouter();
+
+  const handleDelete = async () => {
+    const response = await deleteScanReport(id);
+    if (response) {
+      toast.error(`Failed to delete Scan Report! ${response.errorMessage}`);
```

Comment:
```suggestion
      toast.error(`Failed to delete the Scan Report: ${response.errorMessage}`);
```
Just to make this look a little nicer.